### ISSUE TITLE
AwsIotException when try to access Thing Shadow

### ISSUE THREAD

**Issue by juniorklein** [aws/aws-iot-device-sdk-java#125]
We have an intermittent problem when try to access the thing shadow with de get() method.

```java

Caused by: com.amazonaws.services.iot.client.AWSIotException
	at com.amazonaws.services.iot.client.core.AwsIotCompletion.get(AwsIotCompletion.java:213)
	at com.amazonaws.services.iot.client.shadow.AwsIotDeviceCommand.get(AwsIotDeviceCommand.java:58)
	at com.amazonaws.services.iot.client.shadow.AwsIotDeviceCommandManager.runCommand(AwsIotDeviceCommandManager.java:135)
	at com.amazonaws.services.iot.client.shadow.AwsIotDeviceCommandManager.runCommandSync(AwsIotDeviceCommandManager.java:108)
	at com.amazonaws.services.iot.client.shadow.AbstractAwsIotDevice.get(AbstractAwsIotDevice.java:93)
	at com.amazonaws.services.iot.client.AWSIotDevice.get(AWSIotDevice.java:299)

```

The code is very simple and works well until November/2019 without updates before the problem starts.

```java
AWSIotDevice device = new AWSIotDevice(thingId);
client.attach(device);
String deviceShadow = device.get(SHADOW_TIMEOUT);
```

Enabling a finer logging level, we found a warning saying that a timeout occurred while trying to subscribe to the topics $aws/things/ECFABC1F79ED/shadow/get/accepted and others

```java

2019-11-25 15:01:43.403  WARN 5709 --- [pool-1-thread-1] c.a.s.i.c.s.AwsIotDeviceDeltaListener    : Timeout when subscribing to device topic $aws/things/ECFABC1F79ED/shadow/update/delta
2019-11-25 15:01:43.403  WARN 5709 --- [pool-1-thread-1] a.s.i.c.s.AwsIotDeviceCommandAckListener : Timeout when subscribing to device topic $aws/things/ECFABC1F79ED/shadow/get/accepted
2019-11-25 15:01:43.403  WARN 5709 --- [pool-1-thread-1] a.s.i.c.s.AwsIotDeviceCommandAckListener : Timeout when subscribing to device topic $aws/things/ECFABC1F79ED/shadow/get/rejected
2019-11-25 15:01:43.403  WARN 5709 --- [pool-1-thread-1] a.s.i.c.s.AwsIotDeviceCommandAckListener : Timeout when subscribing to device topic $aws/things/ECFABC1F79ED/shadow/update/accepted
2019-11-25 15:01:43.403  WARN 5709 --- [pool-1-thread-1] a.s.i.c.s.AwsIotDeviceCommandAckListener : Timeout when subscribing to device topic $aws/things/ECFABC1F79ED/shadow/update/rejected
2019-11-25 15:01:43.403  WARN 5709 --- [pool-1-thread-1] a.s.i.c.s.AwsIotDeviceCommandAckListener : Timeout when subscribing to device topic $aws/things/ECFABC1F79ED/shadow/delete/accepted
2019-11-25 15:01:43.403  WARN 5709 --- [pool-1-thread-1] a.s.i.c.s.AwsIotDeviceCommandAckListener : Timeout when subscribing to device topic $aws/things/ECFABC1F79ED/shadow/delete/rejected

```

Anyone can help me with that?

Thanks

**Comment by graebm** [comment]
What is the value of `SHADOW_TIMEOUT`? Is it too small, where a slight change in server behavior could have started this problem? Or is it something reasonable like 10+ seconds?

**Comment by juniorklein** [comment]
@graebm , the SHADOW_TIMEOUT is set to 30 seconds. I think every shadow get must be complete in that time. And more, my service is hosted in EC2 in same region of the broker.

**Comment by graebm** [comment]
What version of the SDK are you using? Have you upgraded it recently? If so, when? Has anything else about your software changed recently? How long was the software working before these occasional failures began popping up?

I'll ask the service team if anything has changed on their end which might account for this

**Comment by juniorklein** [comment]
@graebm, we are using the latest versions. But even upgrade the problems persisted.
In contact with AWS support, we found on logs problems related with limits reached. After that, we developed a feature to create a pool of MQTT clients. That feature removed the most of erros that happening, but now we had identified a new type of error.
In a more granulated log level, we identified for some Things that the system are loading shadow, the follow problem:

`
2019-12-15 09:06:04,393 WARN com.amazonaws.services.iot.client.shadow.AwsIotDeviceCommandManager [pool-47-thread-1] Unknown command received from topic $aws/things/<thing-name>/shadow/get/rejected
2019-12-15 09:06:04,393 WARN com.amazonaws.services.iot.client.shadow.AwsIotDeviceCommandManager [pool-74-thread-1] Unknown command received from topic $aws/things/<thing-name>/shadow/get/rejected
2019-12-15 09:06:04,393 WARN com.amazonaws.services.iot.client.core.AbstractAwsIotClient [MQTT Call: SERVER_COMPATECwZOkkriF] Unexpected message received from topic $aws/things/<thing-name>/shadow/get/rejected
2019-12-15 09:06:04,394 WARN com.amazonaws.services.iot.client.shadow.AwsIotDeviceCommandManager [pool-79-thread-1] Unknown command received from topic $aws/things/<thing-name>/shadow/get/rejected
2019-12-15 09:06:04,395 WARN com.amazonaws.services.iot.client.shadow.AwsIotDeviceCommandManager [pool-84-thread-1] Unknown command received from topic $aws/things/<thing-name>/shadow/get/rejected
`
